eng/common/scripts/Invoke-GitHubAPI.ps1

```diff
@@ -0,0 +1,220 @@
+    -Body ($body | ConvertTo-Json) `
+    -Uri $apiURI `
+    -Headers (Get-Headers -token $token) `
+    -MaximumRetryCount 3
+
+  return $resp
+}
+
+function Invoke-GitHubAPIGet {
+  param (
+    [Parameter(Mandatory = $true)]
+    $apiURI,
+    $token
+  )
+
+  if ($token)
+  {
+    $resp = Invoke-RestMethod `
+      -Method GET `
+      -Uri $apiURI `
+      -Headers (Get-Headers -token $token) `
+      -MaximumRetryCount 3
+  }
+  else {
+    $resp = Invoke-RestMethod `
+      -Method GET `
+      -Uri $apiURI `
+      -MaximumRetryCount 3
+  }
+
+  return $resp
+}
+
+function SplitMembers ($membersString)
+{
+  if (!$membersString) { return $null }
+  return @($membersString.Split(",") | % { $_.Trim() } | ? { return $_ })
+}
+
+function List-PullRequests {
+  param (
+    [Parameter(Mandatory = $true)]
+    $RepoOwner,
+    [Parameter(Mandatory = $true)]
+    $RepoName,
+    [ValidateSet("open","closed","all")]
+    $State = "open",
+    $Head,
+    $Base,
+    [ValidateSet("created","updated","popularity","long-running")]
+    $Sort,
+    [ValidateSet("asc","desc")]
+    $Direction
+  )
+
+  $uri = "$GithubAPIBaseURI/$RepoOwner/$RepoName/pulls"
+  if ($State -or $Head -or $Base -or $Sort -or $Direction) { $uri += '?'}
+  if ($State) { $uri += "state=$State&" }
+  if ($Head) { $uri += "head=$Head&" }
+  if ($Base) { $uri += "base=$Base&" }
+  if ($Sort) { $uri += "sort=$Sort&" }
+  if ($Direction){ $uri += "direction=$Direction&" }
+
+  return Invoke-GitHubAPIGet -apiURI $uri
+}
+
+function Add-IssueComment {
+  param (
+    [Parameter(Mandatory = $true)]
+    $RepoOwner,
+    [Parameter(Mandatory = $true)]
+    $RepoName,
+    [Parameter(Mandatory = $true)]
+    $IssueNumber,
+    [Parameter(Mandatory = $true)]
+    $Comment,
+    [Parameter(Mandatory = $true)]
+    $AuthToken
+
+  )
+  $uri = "$GithubAPIBaseURI/$RepoOwner/$RepoName/issues/$IssueNumber/comments"
+
+  $parameters = @{
+    body = $Comment
+  }
+
+  return Invoke-GitHubAPIPost -apiURI $uri -body $parameters -token $AuthToken
+}
+
+# Will add labels to existing labels on the issue
+function Add-IssueLabels {
+  param (
+    [Parameter(Mandatory = $true)]
+    $RepoOwner,
+    [Parameter(Mandatory = $true)]
+    $RepoName,
+    [Parameter(Mandatory = $true)]
+    $IssueNumber,
+    [ValidateNotNullOrEmpty()]
+    [Parameter(Mandatory = $true)]
+    $Labels,
+    [Parameter(Mandatory = $true)]
+    $AuthToken
+  )
+
+  $uri = "$GithubAPIBaseURI/$RepoOwner/$RepoName/issues/$IssueNumber/labels"
+  $labelAdditions = SplitMembers -membersString $Labels
+  $parameters = @{
+    labels = $labelAdditions
+  }
+
+  return Invoke-GitHubAPIPost -apiURI $uri -body $parameters -token $AuthToken
+}
+
+# Will add assignees to existing assignees on the issue
+function Add-IssueAssignees {
+  param (
+    [Parameter(Mandatory = $true)]
+    $RepoOwner,
+    [Parameter(Mandatory = $true)]
+    $RepoName,
+    [Parameter(Mandatory = $true)]
+    $IssueNumber,
+    [ValidateNotNullOrEmpty()]
```

Comment:
Isn't it valid to pass in empty to clear the assignees?

Reply:
Not for this function. That can be done with the Update-Issue function.

Reply:
Not that I think it is important but it would appear as though the Update-Issue function also blocks empty.

Reply:
Oh, I see what you mean. I should check for whitespace in this.
For `Update-Issue` you can pass white space " " to clear labels or assignees.